pkg/apis/workspaces/v1alpha1/union.go

```diff
@@ -9,32 +9,43 @@ type Identified interface {
 	Id() (string, error)
 }
 
+func isNil(v reflect.Value) bool {
+	k := v.Kind()
+	switch k {
+	// IsNil() can be called only on following types
+	case reflect.Chan, reflect.Func, reflect.Interface, reflect.Map, reflect.Slice, reflect.Ptr:
+		return v.IsNil()
+	default:
+		return false
+	}
+}
+
 func visitUnion(union interface{}, visitor interface{}) (err error) {
+
 	visitorValue := reflect.ValueOf(visitor)
 	unionValue := reflect.ValueOf(union)
 	oneMemberPresent := false
 	typeOfVisitor := visitorValue.Type()
 	for i := 0; i < visitorValue.NumField(); i++ {
 		unionMemberToRead := typeOfVisitor.Field(i).Name
 		unionMember := unionValue.FieldByName(unionMemberToRead)
-		if !unionMember.IsNil() {
```

Comment:
In fact the function that should be called instead of `IsNil` is `IsZero`. I already fixed that in [PR #98](https://github.com/devfile/api/pull/98/files#diff-6965f4fa88f2bd3a5d4f7ac21fb9666aR16).